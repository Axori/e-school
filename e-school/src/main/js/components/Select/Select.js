import React from "react";

const Select = ({id, options, selected, onChange}) => {

    const onChangeHandler = (e) => {
        const {value} = e.target;
        const option = options.find(option => option.value === value);
        onChange(option);
    }

    return <select id={id}
                   className="form-select"
                   value={selected?.value}
                   aria-label="Default select example"
                   onChange={onChangeHandler}>
        {options.map(({label, value}) =>
            <option key={value} value={value}>{label}</option>
        )}
    </select>;
}


export default Select;