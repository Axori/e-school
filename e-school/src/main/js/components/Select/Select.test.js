import React from 'react';
import Select from "./Select";
import renderer from 'react-test-renderer';

describe("Select", () => {
    const options = [
        {
            label: "Option 1",
            value: "option1"
        },
        {
            label: "Option 2",
            value: "option2"
        },
        {
            label: "Option 3",
            value: "option3"
        }
    ];
    const defaultProps = {
        id: "select",
        options,
        onChange: jest.fn(),
        selected: options[0].value
    }

    const getComponent = (props = {}) => renderer.create(<Select {...defaultProps} {...props} />)

    it("should render select when options are defined", () => {
        const component = getComponent();

        expect(component.toJSON()).toMatchSnapshot();
    })

    it("should call onChange with selected option item", () => {
        const component = getComponent();

        const [, selectedOption] = options;
        component.root.findByType('select').props.onChange({target: {value: selectedOption.value}})
        expect(defaultProps.onChange).toHaveBeenCalledWith(selectedOption);
    })
})