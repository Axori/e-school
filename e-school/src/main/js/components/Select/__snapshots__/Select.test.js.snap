// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render select when options are defined 1`] = `
<select
  aria-label="Default select example"
  className="form-select"
  id="select"
  onChange={[Function]}
>
  <option
    value="option1"
  >
    Option 1
  </option>
  <option
    value="option2"
  >
    Option 2
  </option>
  <option
    value="option3"
  >
    Option 3
  </option>
</select>
`;
